test/jasmine/tests/pie_test.js

```diff
@@ -183,52 +183,236 @@ describe('Pie traces:', function() {
             values: [2, 2, 2, 2],
-            title: 'Pie',
-            titleposition: 'inhole',
-            hole: 0,
+            title: 'Test<br>Title',
+            titleposition: 'middle center',
+            titlefont: {
+                size: 60
+            },
+            hole: 0.1,
             type: 'pie',
+            textinfo: 'none'
         }], {height: 300, width: 300})
         .then(function() {
             var title = d3.selectAll('.titletext text');
-            expect(title.size()).toBe(0);
+            expect(title.size()).toBe(1);
+            var titleBox = d3.select('g.titletext').node().getBoundingClientRect();
+            var pieBox = d3.select('g.trace').node().getBoundingClientRect();
+            var radius = 0.1 * Math.min(pieBox.width / 2, pieBox.height / 2);
+            var pieCenterPos = getClientPosition('g.trace');
+            // unfortunately boundingClientRect is inaccurate and so we allow an error of 2
+            expect(_verifyPointInCircle(titleBox.left, titleBox.top, pieCenterPos, radius))
+                .toBeLessThan(2);
+            expect(_verifyPointInCircle(titleBox.right, titleBox.top, pieCenterPos, radius))
+                .toBeLessThan(2);
+            expect(_verifyPointInCircle(titleBox.left, titleBox.bottom, pieCenterPos, radius))
+                .toBeLessThan(2);
+            expect(_verifyPointInCircle(titleBox.right, titleBox.bottom, pieCenterPos, radius))
+                .toBeLessThan(2);
         })
         .catch(failTest)
         .then(done);
     });
 
-    it('shows title outside', function(done) {
+    function _verifyTitle(checkLeft, checkRight, checkTop, checkBottom, checkMiddleX) {
+        var title = d3.selectAll('.titletext text');
+        expect(title.size()).toBe(1);
+        var titleBox = d3.select('g.titletext').node().getBoundingClientRect();
+        var pieBox = d3.select('g.trace').node().getBoundingClientRect();
+        // check that margins agree. we leave an error margin of 2.
+        if(checkLeft) expect(Math.abs(titleBox.left - pieBox.left)).toBeLessThan(2);
+        if(checkRight) expect(Math.abs(titleBox.right - pieBox.right)).toBeLessThan(2);
+        if(checkTop) expect(Math.abs(titleBox.top - pieBox.top)).toBeLessThan(2);
+        if(checkBottom) expect(Math.abs(titleBox.bottom - pieBox.bottom)).toBeLessThan(2);
+        if(checkMiddleX) {
+            expect(Math.abs(titleBox.left + titleBox.right - pieBox.left - pieBox.right))
+                .toBeLessThan(2);
+        }
+    }
+
+    it('shows title top center if there is no hole', function(done) {
+        Plotly.newPlot(gd, [{
+            values: [2, 2, 2, 2],
+            title: 'Test<BR>Title',
+            titleposition: 'middle center',
+            titlefont: {
+                size: 12
+            },
+            hole: 0,
+            type: 'pie',
+            textinfo: 'none'
+        }], {height: 300, width: 300})
+        .then(_verifyTitle(false, false, true, false, true))
```

Comment:
`_verifyTitle` is beautifully DRY 🌴 ! But you're running it synchronously here, because the argument to `then` is the *result* of `_verifyTitle`, not a function to call when the `newPlot` promise resolves. Which turns out to be ok, as this plot (with no MathJax or maps) renders synchronously. My preferred way to handle this is to return a function from `_verifyTitle`:

```js
function _verifyTitle(checkLeft, checkRight, checkTop, checkBottom, checkMiddleX) {
    return function() {
        var title = d3.selectAll('.titletext text');
        ...
    };
}
```